java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,274 @@
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.support.design.widget.BottomNavigationView;
+
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.HealthVaultInitializationHandler;
+import com.microsoft.hsg.android.simplexml.HealthVaultSettings;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.RequestCallback;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+
+public class MainActivity
+	extends ListActivity
+	implements HealthVaultInitializationHandler {
+
+	private HealthVaultApp mService;
+	private HealthVaultClient mHVClient;
+	private BottomNavigationView mBottomNav;
+	private LinearLayout mWeightTile;
+
+	private ProgressDialog connectProgressDialog;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+
+		mService = HealthVaultApp.getInstance(this);
+		if (mService.isAppConnected()) {
+			mService.start(this, this);
+		}
+
+		mHVClient = new HealthVaultClient();
+		
+		mBottomNav = (BottomNavigationView) findViewById(R.id.navigation);
+
+		mBottomNav.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
+			@Override
+			public boolean onNavigationItemSelected(MenuItem item) {
+				// handle desired action here
+				// One possibility of action is to replace the contents above the nav bar
+				// return true if you want the item to be displayed as the selected item
+				return true;
+			}
+		});
+
+		mWeightTile = (LinearLayout) findViewById(R.id.weightTile);
+		mWeightTile.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (mService.isAppConnected()) {
+					intent = new Intent(MainActivity.this, WeightActivity.class);
+				}
+				else {
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
+					startActivity(intent);
+				}
+			}
+		});
+	}
+	
+	@Override
+	protected void onStart() {
+		super.onStart();
+		mHVClient.start();
+	}
+	
+	@Override
+    protected void onResume() {
+		super.onResume();
+    }
+	
+    @Override
+	protected void onStop() {
+		super.onStop();
+		if(connectProgressDialog != null) {
+			connectProgressDialog.dismiss();
+		}
+
+		mHVClient.stop();
+	}
+    
+    @Override
+	public boolean onCreateOptionsMenu(Menu menu) {
+		getMenuInflater().inflate(R.menu.main, menu);
+		LinearLayout recordNameLayout = (LinearLayout) menu.findItem(R.id.record_name_layout).getActionView();
+        TextView tv = (TextView) recordNameLayout.findViewById(R.id.current_record_name);
```

Comment:
Lets use more descriptive names instead of "tv", as a sample it will help make it easier for other devs to follow if the name provides a little context.